src/cmd-run

```diff
@@ -0,0 +1,125 @@
+
+set -euo pipefail
+
+VM_DISK=
+VM_MEMORY=2048
+VM_PERSIST=0
+VM_NCPUS="${VM_NCPUS:-$(nproc)}"
+VM_SRV_MNT=
+SSH_PORT=${SSH_PORT:-}
+USAGE="Usage: $0 /path/to/disk.qcow2 [--] [qemu options...]
+Options:
+    -d DISK     Root disk drive (won't be changed by default)
+    --persist   Don't create a temporary snapshot
+    -i FILE     File containing an Ignition config
+    --srv src   Mount (via 9p) src on the host as /var/srv in guest
+    -m MB       RAM size in MB (2048)
+    -p PORT     The port on localhost to map to the VM's sshd. [2222]
+    -h          this ;-)
+
+This script is a wrapper around qemu for starting CoreOS virtual machines,
+it will auto-log you into the console, and by default for read-only disk
+images makes a transient snapshot.
+
+Any arguments after -a and -p will be passed through to qemu, -- may be
+used as an explicit separator. See the qemu(1) man page for more details.
+"
+
+die(){
+	echo "${1}" 1>&2
+	exit 1
+}
+
+while [ $# -ge 1 ]; do
+    case "$1" in
+        -d)
+            VM_DISK="$2"
+            shift 2 ;;
+        --persist)
+            VM_PERSIST=1
+            shift 1 ;;
+        -i|-ignition-config)
+            IGNITION_CONFIG_FILE="$2"
+            shift 2 ;;
+        --srv)
+            VM_SRV_MNT="$2"
+            shift 2 ;;
+        -m)
+            VM_MEMORY="$2"
+            shift 2 ;;
+        -p|-ssh-port)
+            SSH_PORT="$2"
+            shift 2 ;;
+        -v|-verbose)
+            set -x
+            shift ;;
+        -h|-help|--help)
+            echo "$USAGE"
+            exit ;;
+        --)
+            shift
+            break ;;
+        *)
+            break ;;
+    esac
+done
+
+if [ -z "${VM_DISK}" ]; then
+    if [ -L ./builds/latest ]; then
+        VM_DISK=$(ls builds/latest/*-qemu.qcow2)
+    else
+        die "No builds/latest, and no -d argument provided"
+    fi
+fi
+
+# Emulate the host CPU closely in both features and cores.
+set -- -machine accel=kvm -cpu host -smp "${VM_NCPUS}" "$@"
+
+if [ -n "${VM_SRV_MNT}" ]; then
+    set -- --fsdev local,id=var-srv,path=${VM_SRV_MNT},security_model=mapped,readonly "$@"
+    ign_var_srv_mount=',{
+"name": "var-srv.mount",
+"enabled": true,
+"contents": "[Mount]\nWhat=/var/srv\nWhere=/var/srv\nType=9p\nOptions=ro,trans=virtio,version=9p2000.L\n[Install]\nWantedBy=multi-user.target\n"
+}'
+else
+    ign_var_srv_mount=""
+fi
+
+if [ -z "${IGNITION_CONFIG_FILE:-}" ]; then
+    f=$(mktemp)
+    cat > ${f} <<EOF
+{"ignition": {"config": {}, "security": {"tls": {}}, "timeouts": {}, "version": "2.2.0"}, "networkd": {}, "passwd": {"users": [{"groups": ["sudo"], "name": "core"}]}, "storage": {}, "systemd": {"units": [{"name": "serial-getty@ttyS0.service", "dropins": [{"name": "autologin-core.conf", "contents": "[Service]\nTTYVTDisallocate=no\nExecStart=\nExecStart=-/usr/sbin/agetty --autologin core --noclear %I \$TERM\n"}]}${ign_var_srv_mount}]}}
+EOF
+    exec 3<>${f}
+    if ! jq . < ${f} >/dev/null; then
```

Comment:
Probably worth adding `ignition-validate` to the container here for this.

Reply:
Mind if that's a separate PR?

Reply:
Yup, sure thing!